sdk/test-utils/recorder/src/utils/utils.ts

```diff
@@ -338,6 +339,65 @@ export function isPlaybackMode() {
   return !isRecordMode() && !isLiveMode();
 }
 
+export function resolveAssetsJson(testPath: string): string {
+  if (!(typeof testPath === "undefined" || testPath == null)) {
+    if (process === undefined) {
+      throw new RecorderError(
+        "Unable to access the project object. This is likely occurring because the user requested a browser test" +
+          " run and RECORDING_ASSETS_PATH isn't yet defined. Set process.env variable RECORDING_ASSETS_PATH within karma configuration."
+      );
+    }
+
+    const currentPath = process.cwd();
+
+    let rootPath = undefined;
+
+    if (fs.existsSync(path.join(currentPath, "package.json"))) {
+      // <root>/sdk/service/project/package.json
+      if (fs.existsSync(path.join(currentPath, "package.json"))) {
+        const expectedRootPath = path.join(currentPath, "..", "..", ".."); // <root>/
+        if (
+          fs.existsSync(path.join(expectedRootPath, "sdk/")) && // <root>/sdk
+          fs.existsSync(path.join(expectedRootPath, "rush.json")) // <root>/rush.json
+        ) {
+          // reached root path
+          rootPath = expectedRootPath;
+        }
+      }
+    } else {
+      throw new RecorderError(`'package.json' is not found at ${currentPath}`);
+    }
+
+    if (rootPath === undefined) {
+      throw new RecorderError(
+        "rootPath or expectedProjectPath could not be calculated properly from process.cwd()"
+      );
+    }
+
+    let targetPath: string = path.dirname(path.join(rootPath, testPath));
```

Comment:
q: Is there any reason we can't just start from `process.cwd()` here like `relativeRecordingsPath()` does? Seems like that would simplify things quite a bit, but maybe there's an edge case I didn't consider.

```suggestion
    const targetPath = process.cwd();
```

Reply:
You would know better than me. The way I think about it, if I accept a virtual `testPath`, on `node` runs that will resolve out to a path that doesn't technically exist. eg:

`sdk/textanalytics/ai-text-analytics/recordings/node/aad_textanalyticsclient_lros_analyze/recording_all_documents_with_errors_and_multiple_actions.json`

Is there a case where using `process.cwd` could mean possibly not assending through that virtual path until we hit an `assets.json`?

```
sdk/textanalytics/
  <tests invoked from here>
  ai-text-analytics/
    assets.json
  assets.json
```

I would definitely encourage against this scenario, but it's technically possible. I feel like I'm tying myself in knots to explain what I'm afraid of. Will commit your suggestion :P